File: app/src/main/java/suzp1984/github/io/exapidemo/app/design/NavgationViewActivity.java
package suzp1984.github.io.exapidemo.app.design;

import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.Gravity;
import android.view.View;

import butterknife.BindView;
import butterknife.ButterKnife;
import suzp1984.github.io.exapidemo.R;

public class NavgationViewActivity extends AppCompatActivity {

    @BindView(R.id.recyclor)
    RecyclerView mRecycler;

    @BindView(R.id.toolbar)
    Toolbar mToolbar;

    @BindView(R.id.drawer_layout)
    DrawerLayout mDrawer;

    @BindView(R.id.menu_list)
    RecyclerView mMenuList;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_navgation_view);

        ButterKnife.bind(this);

        if (mToolbar != null) {
            setSupportActionBar(mToolbar);
            mToolbar.setNavigationIcon(R.mipmap.ic_launcher);
            mToolbar.setNavigationOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    mDrawer.openDrawer(Gravity.LEFT);
                }
            });
        }

        mRecycler.setAdapter(new HeaderAdapter());
        mRecycler.setLayoutManager(new LinearLayoutManager(this));

        mMenuList.setAdapter(new HeaderAdapter());
        mMenuList.setLayoutManager(new LinearLayoutManager(this));
    }
}

File: app/src/main/kotlin/suzp1984/github/io/exapidemo/app/design/NavgationViewActivity.kt
package suzp1984.github.io.exapidemo.app.design

import android.support.v4.widget.DrawerLayout
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.Toolbar
import android.view.Gravity
import android.view.View

import butterknife.BindView
import butterknife.ButterKnife
import suzp1984.github.io.exapidemo.R

class NavgationViewActivity : AppCompatActivity() {

    @BindView(R.id.recyclor)
    internal lateinit var mRecycler: RecyclerView

    @BindView(R.id.toolbar)
    internal lateinit var mToolbar: Toolbar

    @BindView(R.id.drawer_layout)
    internal lateinit var mDrawer: DrawerLayout

    @BindView(R.id.menu_list)
    internal lateinit var mMenuList: RecyclerView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_navgation_view)

        ButterKnife.bind(this)

        setSupportActionBar(mToolbar)
        mToolbar.setNavigationIcon(R.mipmap.ic_launcher)
        mToolbar.setNavigationOnClickListener { mDrawer.openDrawer(Gravity.LEFT) }

        mRecycler.adapter = HeaderAdapter()
        mRecycler.layoutManager = LinearLayoutManager(this)

        mMenuList.adapter = HeaderAdapter()
        mMenuList.layoutManager = LinearLayoutManager(this)
    }
}
